subsys/input/input.c

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/input/input.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(input, CONFIG_INPUT_LOG_LEVEL);
+
+#define INPUT_TYPE_MASK 0x7fff
+
+#define INPUT_TYPE(t) (t & INPUT_TYPE_MASK)
+
+#define INPUT_SYN_IS_SET(v) (v & INPUT_EV_SYN)
+
+#ifdef CONFIG_INPUT_MODE_THREAD
+
+K_MSGQ_DEFINE(input_msgq, sizeof(struct input_event),
+	      CONFIG_INPUT_QUEUE_MAX_MSGS, 4);
+
+#endif
+
+static void input_process(struct input_event *evt)
+{
+	bool sync = INPUT_SYN_IS_SET(evt->type);
+
+	evt->type = INPUT_TYPE(evt->type);
```

Comment:
This only removes sync from the type field, which looks like a workaround.

Reply:
It's not but I got the point.